model/provider.go

```diff
@@ -0,0 +1,22 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
```

Comment:
Please don't call your package `model`. Read more about package by layer vs by feature in Uncle Bob's Clean Code book.

Reply:
`types` is a good alternative

Reply:
No, it's not. :D

This should be in the provider package.

Package by layer is not a very good architectural approach.

Reply:
You lost me @sagikazarmark 😮‍💨
I would emphasise that this is an MVP and we do not need to focus too much on issues like these. I'd leave it as is for now (or if you have a suggestion on what the naming should be, shoot). Otherwise, I am okay with this for now.

Reply:
> otherwise we have `provider.Provider`

I don't have a problem with that. (BTW we don't actually need this interface apart from the place in code where we consume it)


> I would emphasise that this is an MVP and we do not need to focus too much on issues like these

I disagree. This is a project where Bence should learn the good practices, not the bad ones. Package by layer, especially here is bad.

Reply:
To wrap up into action items:
- Move `NewProvider` constructor back to main package
- Move back the `Provider` interface to `provider` package (this removes the `models` package)